### PR TITLE
Always keep track of latest offset

### PR DESCRIPTION
**Pull request by arj03**
As it is now, when the normal log finishes processing, all indexes will get a [write](https://github.com/ssb-ngi-pointer/ssb-db2/blob/05a7779953c22fae7846c05e113d5fb16b022c28/db.js#L263). If an index doesn't write something for each message, lets say its only interested in a particular type, then the last write might not write the final offset which means that you can't do something like onDrain on that index. Furthermore when you restart the app, it has to go through all those messages again it just checked. In the live case we have the same problem, but here we don't want to write for each message so we debounce the write a bit.

### CODE REVIEW

**Comment by arj03** [comment]
I just did a full sync in the browser with this and I don't have any problems with the browser getting slow after initial sync, so it seems like this is the last part I need in order to be able to merge in the ssb-db2 changes :-)

**Comment by arj03** [comment]
@staltz I'm not sure if you are using indexes in manyverse yet. But the problem I was experiencing was that if the indexes catches up with the log, then they will go into live mode and before this would write for every message :(

**Comment by staltz** [comment]
Good catch! I'm not using these indexes yet in Manyverse, it's useful you fixed this at this point.

File: indexes/plugin.js [hunk 1]
```diff
@@ -51,13 +52,15 @@ module.exports = function (
     } else cb()
   }
 
+  const liveWriteBatch = debounce(writeBatch, 250)
+
   function onData(record, isLive) {
-    // maybe check if for us!
     const unwritten = handleData(record, processed)
-    if (unwritten > 0) unWrittenOffset = record.offset
+    unWrittenOffset = record.offset
```
Comment (staltz) [inline indexes/plugin.js:59]: Why was the if removed?

Comment (arj03) [reply indexes/plugin.js:59]: That is this part: If an index doesn't write something for each message

Comment (staltz) [reply indexes/plugin.js:59]: Ah I see

**Review by staltz — APPROVED** [review]

**Pull request merged**